src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Helpers/DelegationHelper.cs

```diff
@@ -275,5 +275,51 @@ public static string GetAttributeMatchKey(List<AttributeMatch> attributeMatches)
         {
             return string.Concat(attributeMatches.OrderBy(r => r.Id).Select(r => r.Id + r.Value));
         }
+
+        /// <summary>
+        /// Sets the RuleType on each rule in the given list
+        /// </summary>
+        public static void SetRuleType(List<Rule> rulesList, int offeredByPartyId, List<int> keyRolePartyIds, List<AttributeMatch> coveredBy, int parentPartyId = 0)
+        {
+            foreach (Rule rule in rulesList)
+            {
+                if (TryGetDelegationParamsFromRule(rule, out _, out _, out _, out int? coveredByPartyId, out int? coveredByUserId, out _)
+                    && rule.Type == RuleType.None)
+                {
+                    SetTypeForSingleRule(keyRolePartyIds, offeredByPartyId, coveredBy, parentPartyId, rule, coveredByPartyId, coveredByUserId);
+                }
+            }
+        }
+
+        private static void SetTypeForSingleRule(List<int> keyRolePartyIds, int offeredByPartyId, List<AttributeMatch> coveredBy, int parentPartyId, Rule rule, int? coveredByPartyId, int? coveredByUserId)
+        {
+            bool isUserId = TryGetCoveredByUserIdFromMatch(coveredBy, out int coveredByUserIdFromRequest);
+            bool isPartyId = TryGetCoveredByPartyIdFromMatch(coveredBy, out int coveredByPartyIdFromRequest);
+
+            if (((isUserId && coveredByUserIdFromRequest == coveredByUserId) || (isPartyId && coveredByPartyIdFromRequest == coveredByPartyId))
+                && rule.OfferedByPartyId == offeredByPartyId)
+            {
+                if (parentPartyId != 0 && parentPartyId == rule.OfferedByPartyId && rule.OfferedByPartyId == offeredByPartyId)
+                {
+                    rule.Type = RuleType.InheritedAsSubunit;
+                }
+                else
+                {
+                    rule.Type = RuleType.DirectlyDelegated;
+                }
+            }
+            else if (isUserId && keyRolePartyIds.Any(id => id == coveredByPartyId) && rule.OfferedByPartyId == offeredByPartyId)
+            {
+                rule.Type = RuleType.InheritedViaKeyRole;
+            }
+            else if (isPartyId && keyRolePartyIds.Any(id => id == coveredByPartyId) && rule.OfferedByPartyId == offeredByPartyId)
+            {
```

Comment:
InheritedAsSubunitViaKeyrole is when:
if (isUserId && keyRolePartyIds.Any(id => id == coveredByPartyId) && rule.OfferedByPartyId = parentPartyId)

Reply:
Fixed